### PR TITLE
Associations

### PR DESCRIPTION
**Pull request by bmneely**
@robmadden want to take a look

### CODE REVIEW

File: lib/bloc_record/persistence.rb [hunk 1]
```diff
@@ -12,10 +12,10 @@ def create(attrs)
       attrs.delete "id"
       vals = attributes.map { |key| BlocRecord::Utility.sql_strings(attrs[key]) }
 
-      connection.execute <<-SQL
-        INSERT INTO #{table} (#{attributes.join ","})
-        VALUES (#{vals.join ","});
-      SQL
+      connection.execute(
```
Comment (robmadden) [inline lib/bloc_record/persistence.rb]: I like this syntax better, seems cleaner than the `<<-SQL`. :+1:

File: lib/bloc_record/persistence.rb [hunk 2]
```diff
@@ -25,7 +25,7 @@ def create(attrs)
     def update(ids, updates)
       updates = BlocRecord::Utility.convert_keys(updates)
       updates.delete "id"
-      updates_array = updates.map { |key, value| "#{key}=#{BlocRecord::Utility.sql_strings(value)}" }
+      updates_array = updates.keys.map { |key| "#{key} = ?" }
```
Comment (robmadden) [inline lib/bloc_record/persistence.rb]: So much better :+1:

File: lib/bloc_record/associations.rb [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+  def has_many(association)
+    self.send(:define_method, association) do
+      rows = self.class.connection.execute <<-SQL
+        SELECT * FROM #{association.to_s.singularize}
+        WHERE #{self.class.table}_id = #{self.id}
+      SQL
+
+      class_name = association.to_s.classify.constantize
+      collection = BlocRecord::Collection.new
+      rows.each { |row| collection << class_name.new(Hash[class_name.columns.zip(row)]) }
```
Comment (robmadden) [inline lib/bloc_record/associations.rb]: I think we should break this line up a bit.

File: lib/bloc_record/associations.rb [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+
+      class_name = association_name.classify.constantize
+
+      if row
+        data = Hash[class_name.columns.zip(row)]
+        class_name.new(data)
+      end
+    end
+  end
+end
```
Comment (robmadden) [inline lib/bloc_record/associations.rb]: So we're fetching the first row, and creating an instance of the class with the column names from that row? Just want to be sure I caught the logic correctly here.

Comment (bmneely) [inline lib/bloc_record/associations.rb]: Yep, we're just fetching the parent association and creating an instance.

File: lib/bloc_record/associations.rb (hunk 3 above)
Comment (bmneely) [inline lib/bloc_record/associations.rb]: What?! Really? I was thinking we should make it more longer!

**Pull request merged**